rpc-rs/src/provider.rs

```diff
@@ -1,4 +1,5 @@
 #![cfg(not(target_arch = "wasm32"))]
+#![allow(clippy::let_and_return)] // fixes some false-positive clippy warnings with feature flags
```

Comment:
Nit: I generally like adding the allows exactly where they are needed unless we are talking about something more than 3 of these similar lints

Reply:
me too ... but I removed this line and don't get clippy warnings any more so it's out!